app/src/main/java/de/danoeh/antennapod/UpdateManager.java

```diff
@@ -0,0 +1,86 @@
+package de.danoeh.antennapod;
+
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.util.Log;
+
+import java.io.File;
+import java.util.List;
+
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedImage;
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.DBWriter;
+
+public class UpdateManager {
+
+    public static final String TAG = UpdateManager.class.getSimpleName();
+
+    private static final String PREF_NAME = "app_version";
+    private static final String KEY_VERSION_CODE = "version_code";
+
+    private static int currentVersionCode;
+
+    private static Context context;
+    private static SharedPreferences prefs;
+
+    public static void init(Context context) {
+        UpdateManager.context = context;
+        prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        PackageManager pm = context.getPackageManager();
+        try {
+            PackageInfo info = pm.getPackageInfo(context.getPackageName(), 0);
+            currentVersionCode = info.versionCode;
+        } catch (PackageManager.NameNotFoundException e) {
+            Log.e(TAG, "Failed to obtain package info for package name: " + context.getPackageName(), e);
+            currentVersionCode = 0;
+        }
+        final int oldVersionCode = getStoredVersionCode();
+        Log.d(TAG, "old: " + oldVersionCode + ", current: " + currentVersionCode);
+        if(oldVersionCode < currentVersionCode) {
+            onUpgrade(oldVersionCode, currentVersionCode);
+            setCurrentVersionCode();
+        }
+    }
+
+    public static int getCurrentVersionCode() {
+        return currentVersionCode;
+    }
+
+    public static int getStoredVersionCode() {
+        return prefs.getInt(KEY_VERSION_CODE, -1);
+    }
+
+    public static void setCurrentVersionCode() {
+        prefs.edit().putInt(KEY_VERSION_CODE, currentVersionCode).apply();
+    }
+
+    private static void onUpgrade(final int oldVersionCode, final int newVersionCode) {
+        if(oldVersionCode < 1030000) {
```

Comment:
I think the version code check here needs to be higher. Otherwise people on the alpha won't have this cleanup done.